KAFKA-18597 Fix max-buffer-utilization-percent is always 0 for 3.9

LoganZhuZzz opened the pull request:
This patch is intended for the 3.9 branch and resolves the issue where max-buffer-utilization-percent was always returning 0. 

During the process, merge conflicts were resolved to ensure compatibility with the 3.9 branch. 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

LoganZhuZzz committed:
KAFKA-18597: Fix max-buffer-utilization-percent is always 0

frankvicky committed:
Clean-up LogCleaner#maxOverCleanerThreads

LoganZhuZzz committed:
KAFKA-18597: Fix test issue

LoganZhuZzz committed:
KAFKA-18597: Add unit test

LoganZhuZzz committed:
KAFKA-18597: Add unit tests for additional impacted metrics in LogCleanerTest

- Added testMaxBufferUtilizationPercentMetric to validate maximum buffer utilization metric.
- Added testMaxCleanTimeMetric to verify the maximum clean time metric.
- Added testMaxCompactionDelayMetrics to ensure correctness of maximum compaction delay metric.

LoganZhuZzz committed:
KAFKA-18597: Fix test error

chia7712 commented on core/src/main/scala/kafka/log/LogCleaner.scala:
```diff
@@ -116,16 +116,15 @@ class LogCleaner(initialConfig: CleanerConfig,
-  private def maxOverCleanerThreads(f: CleanerThread => Double): Int =
-    cleaners.foldLeft(0.0d)((max: Double, thread: CleanerThread) => math.max(max, f(thread))).toInt
+  private[log] def maxOverCleanerThreads(f: CleanerThread => Double): Double =
+    cleaners.map(f).maxOption.getOrElse(0.0d)
```
3.9 branch must support scala 2.12, so we don't use `maxOption`

LoganZhuZzz replied:
Hi, @chia7712 . Thank you for the feedback! I’ve ensured that all changes are compatible with Scala 2.12 as required for the 3.9 branch. This includes:
- Replacing maxOption with foldLeft for compatibility.
- Avoiding Scala 2.13-specific features like underscore separators in numeric literals.
- Ensuring test cases and implementation align with Scala 2.12 syntax and functionality.

Let me know if there’s anything else you’d like me to adjust!

LoganZhuZzz committed:
KAFKA-18597: add support for Scala 2.12 compatibility

- Replaced `maxOption` with `foldLeft` for compatibility with Scala 2.12.
- Removed underscore separators in numeric literals to align with Scala 2.12 syntax.
